ffsync/memory_locker.go

```diff
@@ -140,18 +146,25 @@ func (m *memoryLocker) Unlock(key Key) error {
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
 
-	lockInfo, ok := m.lockedItems[key.Key()]
+	lockInfo, ok := m.lockedItems.Load(key.Key())
 	if !ok {
 		return fferr.NewKeyNotLockedError(key.Key(), nil)
 	}
 
-	if lockInfo.ID != key.ID() {
-		err := fferr.NewKeyAlreadyLockedError(key.Key(), lockInfo.ID, fmt.Errorf("attempting to unlock with incorrect key"))
-		err.AddDetail("expected key", lockInfo.ID)
+	keyLock, ok := lockInfo.(LockInformation)
+	if !ok {
+		return fferr.NewInternalError(fmt.Errorf("could not cast lock information"))
+	}
+
+	if keyLock.ID != key.ID() {
+		err := fferr.NewKeyAlreadyLockedError(key.Key(), keyLock.ID, fmt.Errorf("attempting to unlock with incorrect key"))
+		err.AddDetail("expected key", keyLock.ID)
 		err.AddDetail("received key", key.ID())
 		return err
 	}
-	delete(m.lockedItems, key.Key())
+
+	m.lockedItems.Delete(key.Key())
+
```

Comment:
down below you close the done channel. Just want to ensure close can't be called on an already close channel, i think it'll panic if that happens

Reply:
@ahmadnazeri

Reply:
@ahmadnazeri